### ISSUE TITLE
Redux wallet state diverges from context

### ISSUE THREAD

**Issue by woodenfurniture** [shapeshift/web#7904]
### Overview

When switching wallets, the redux state differs from the state in the context returned by `useWallet`.

### References and additional details

When changing wallet, the following return different results:

```typescript
  const walletId = useAppSelector(selectWalletId)

  const {
    state: { deviceId: walletDeviceId },
  } = useWallet()
```

### Acceptance Criteria

The wallet state held by redux should be identical to the state returned by `useWallet`.

**Comment by woodenfurniture** [comment]
The issue can be repro'd with the following steps:
1. Open app. Redux and context device IDs dont match - ignore this for now.
2. Connect native wallet. Redux and context match.
3. Click "switch wallet provider" and click a different native wallet without entering the password. Wallet provider changes device ID eagerly. Redux device ID is now different to wallet provider context.
4. Exit the "switch wallet provider" UI. Wallet device ID is still different. Wallet context now has the wrong device ID and redux has the correct one until user reloads page or switches to a different wallet.

Basically the state changes in the wallet provider are incorrect/incomplete resulting in the ability to get into a state where the device ID is incorrect. I think the root cause is the wallet context using the deviceId to store a "pending" device ID during the connection phase of switching device, resulting in corruption when a user backs out of switching device.

**Comment by woodenfurniture** [comment]
![image](https://github.com/user-attachments/assets/a598cd9a-ff35-4de5-bca4-0af4f34b86d0)

pPlan of action:
1. stop the bleeding - ensure the wallet provider device ID is cleared when user backs out of connecting the wallet so it doesnt persist with wrong ID (see image above)
2. deduplicate the wallet device id in redux (i.e remove portfolio slice wallet meta and use localWalletSlice instead)
3. add notion of "pending" wallet state to wallet provider
4. rename local wallet state to somehow indicate "last connected" to better represent what it's for
5. ensure localWalletSlice is always slaved to the wallet provider in all cases
6. re-test and look for more bugs

**Comment by woodenfurniture** [comment]
Point 1) not doable because the device ID is tied to the connected wallet, meaning clearing the pending device ID when a user backs out introduces a different issue - the connected device ID will be undefined rather than an incorrect string. 

After attempting to introduce a `pendingDeviceId`, it seems more likely we need the concept of a `pendingWallet: WalletInfo | null` which stores the entire state of the wallet about to be connected. Upon connecting, this will replace `walletInfo`.

Digging further, its also likely we'd need to rework the wallet states to transition from pending to connected state:

- `WalletActions.SET_WALLET` currently handles both "pending" and "connected" states. 
- `WalletActions.SET_IS_CONNECTED` is then used ot mark the pending state as connected, but if a user backs out of connecting, the state is irrecoverably corrupted until they reload the page.

We'd be better off reworking the state transitions to support canceling:
- `WalletActions.SET_PENDING_WALLET_INFO` to store the pending state of the wallet being connected
- `WalletActions.SET_IS_CONNECTED` copies the `pendingWallet` state to `walletInfo`
- `WalletActions.UNSET_PENDING_WALLET_INFO` to set the pending wallet state back to null when a user cancels

Noting the large expansion in scope here, I'll discuss with the team before proceeding.

**Comment by Olowojaye** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by woodenfurniture** [comment]
Fix went stale in review so will have to re-author
https://github.com/shapeshift/web/pull/7937#issuecomment-2445423945

**Comment by 0xean** [comment]
@woodenfurniture  to pick this back up after limit orders, its a rare edge case scenario that we have no reported issues from in prod.
